specutils/spectra/spectral_axis.py

```diff
@@ -0,0 +1,78 @@
+import warnings
+
+import astropy.units as u
+from astropy.utils.decorators import lazyproperty
+import numpy as np
+
+from .spectral_coordinate import SpectralCoord
+
+
+__all__ = ['SpectralAxis']
+
+# We don't want to run doctests in the docstrings we inherit from Quantity
+__doctest_skip__ = ['SpectralAxis.*']
+
+class SpectralAxis(SpectralCoord):
+    """
+    Coordinate object representing spectral values corresponding to a specific
+    spectrum. Overloads SpectralCoord with additional information (currently
+    only bin edges).
+
+    Parameters
+    ----------
+    bin_specification: str, optional
+        Must be "edges" or "centers". Determines whether specified axis values
+        are interpreted as bin edges or bin centers. Defaults to "centers".
+    """
+
+    def __new__(cls, value, *args, bin_specification="centers", **kwargs):
+
+        # Convert to bin centers if bin edges were given, since SpectralCoord
+        # only accepts centers
+        if bin_specification == "edges":
+            bin_edges = value
```

Comment:
```suggestion
    diff_val = np.diff(value)
    
    if not np.allclose(diff_val, diff_val[0], atol=0.001):
        warnings.warn("Edge-based spectral axis does not seem to be linear. Conversions to bin centers may be incorrect.", AstropyUserWarning)
```

Reply:
@eteq Would this make sense as a temporary solution before we consider dealing with non-linear spectral axis conversions?

Reply:
Actually I think this warning is unnecessary.  non-linear spectral axes are quite common, but it's okay to assume a convention in this case and not bother with the warning.